docs/acn-internals.md

```diff
@@ -0,0 +1,166 @@
+
+The aim of this document is to describe at a high-level
+the main implementation of the Agent Communication Network (ACN).
+
+## Introduction
+
+The ACN protocol implements transmission control over the ACN.
+The message serialization is based on 
+<a href="https://developers.google.com/protocol-buffers" target="_blank">Protocol Buffers</a>.
+and the definition of the data structures involved is defined
+<a href="https://github.com/fetchai/agents-aea/blob/develop/libs/go/libp2p_node/acn/acn_message.proto" target="_blank">here</a>.
+
+## Messages
+
+### Agent Record
+
+An _Agent Record_ is a data structure containing information about an agent connected to the ACN.
+
+> TODO: list and explain fields types and purpose
+
+### ACN Message
+
+Entities in the ACN (i.e. either agents or peers) exchange _ACN messages_.
+An ACN message contains the following fields:
+
+- _version_: a string that stores the protocol version, following <a href="https://semver.org/" target="_blank">Semantic Versioning</a>.
```

Comment:
See my branch https://github.com/fetchai/agents-aea/pull/2447, I am not sure there is any point in adding this. Either the protocol stays the same and is compatible, or we have a breaking change...

Reply:
Also check out this branch for the full specification.